internal/streams/streams.go

```diff
@@ -50,13 +50,8 @@ func streamFromBucket(bucket, streamInBucket int) int {
 }
 
 func (s *IDGenerator) GetStream() (int, bool) {
-	// based closely on the java-driver stream ID generator
-	// avoid false sharing subsequent requests.
-	offset := atomic.LoadUint32(&s.offset)
-	for !atomic.CompareAndSwapUint32(&s.offset, offset, (offset+1)%s.numBuckets) {
-		offset = atomic.LoadUint32(&s.offset)
-	}
-	offset = (offset + 1) % s.numBuckets
+	// Reduce collision by offsetting the starting point
+	offset := atomic.AddUint32(&s.offset, s.numBuckets/10+1)
```

Comment:
why s.numBuckets/10?

Reply:
Devisor should be big enough to provide some cycles before reaching numbucket, and small enough to have significant gap between invocation to ensure that collision chance is actually reduced.
And resulted number should decide numbuckets with some remainder, to ensure it does not land on the same indexes all the time.

Let me put it into comment and I think we need to have more logic on how to pick correct offest

Reply:
Please take a look